src/rebase-snap/src/main.rs

```diff
@@ -43,7 +43,29 @@ fn build_arg_parser<'a>() -> ArgParser<'a> {
     arg_parser
 }
 
-fn extract_args<'a>(arg_parser: &'a mut ArgParser<'a>) -> &'a Arguments<'a> {
+#[derive(Debug)]
+struct RsExitError {
+    exit_code: ExitCode,
+    error_msg: String,
+}
+
+impl Termination for RsExitError {
+    fn report(self) -> ExitCode {
+        eprintln!("{}", self.error_msg);
+        self.exit_code
+    }
+}
+
+impl RsExitError {
+    fn new(exit_code: ExitCode, error_msg: String) -> Self {
+        RsExitError {
+            exit_code,
+            error_msg,
+        }
+    }
+}
+
+fn extract_args<'a>(arg_parser: &'a mut ArgParser<'a>) -> Result<&'a Arguments<'a>, RsExitError> {
```

Comment:
This should return a custom error type (same comment as on `run_with_api` https://github.com/firecracker-microvm/firecracker/pull/3575/commits/f83b7c8cd3fdb3b320ee1fe1f80769c4c498ee56#r1163827837).